### PR TITLE
chore: CI - add env picker for publishing experiments

### PR DESCRIPTION
**Pull request by metal-messiah**
Please add a one-paragraph summary here, suitable for a release notes description. This will help with documentation.
---

### CODE REVIEW

**Review by patrickhousley — COMMENTED** [review]
We should put this in a sub-directory of experiments in the S3 bucket. When a PR is merged to main, the publish workflow will rebuild both dev and stage which will include all the experiments anyways.

File: .github/workflows/publish-experiment.yml
Comment (cwli24) [inline .github/workflows/publish-experiment.yml:1]: Suggestion to combine the two jobs into one instead:
```javascript
# This workflow can be ran on any branch to build and publish
# that branch as an internal experiment. Once published, the
# dev and staging A/B scripts will be updated to include the
# new experiment.
name: Publish Experiment

on:
  workflow_dispatch:
    inputs:
      env:
        type: choice
        description: The env to publish the experiment to
        options:
          - dev
          - staging
        required: true

jobs:
  # Build and publish the branch to S3 experiments folder
  publish-experiment-to-s3:
    runs-on: ubuntu-latest
    timeout-minutes: 30
    defaults:
      run:
        shell: bash
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: lts/*
      - name: Install project dependencies
        run: npm ci
      - name: Clean branch name
        id: clean-branch-name
        run: echo "results=$(echo '${{ github.ref }}' | sed 's/refs\/heads\///g' | sed 's/[^[:alnum:].-]/-/g')" >> $GITHUB_OUTPUT
      - name: Build experiment
        run: npm run cdn:build:experiment -- --env branchName='${{ steps.clean-branch-name.outputs.results }}'
      - name: Upload experiment
        id: s3-upload
        uses: ./.github/actions/s3-upload
        with:
          aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws_role: ${{ secrets.AWS_ROLE_ARN }}
          aws_bucket_name: ${{ secrets.AWS_BUCKET }}
          local_dir: $GITHUB_WORKSPACE/build
          bucket_dir: experiments/${{ steps.clean-branch-name.outputs.results }}
      - name: Gather purge paths
        id: purge-paths
        run: echo "results=$(echo '${{ steps.s3-upload.outputs.results }}' | jq -j '.[].Key | select(. | test("nr-loader-.*?\\.js$") or test(".*stats\\.json$")) + " "')" >> $GITHUB_OUTPUT
      - name: Purge fastly cache
        uses: ./.github/actions/fastly-purge
        with:
          fastly_key: ${{ secrets.FASTLY_PURGE_KEY }}
          fastly_service: js-agent.newrelic.com
          purge_path: ${{ steps.purge-paths.outputs.results }}

  # Rebuild and publish the chosen environment A/B script
  publish-ab:
    needs: [publish-experiment-to-s3]
    runs-on: ubuntu-latest
    timeout-minutes: 30
    defaults:
      run:
        shell: bash
    steps:
      - uses: actions/checkout@v3
        with:
          ref: main
      - uses: actions/setup-node@v3
        with:
          node-version: lts/*
      - name: Determine environment app id
        id: local-vars
        env: 
            INPUT_ENV: ${{ github.event.inputs.env }}
        run: |
            echo "::set-output name=app_id_key::INTERNAL_${INPUT_ENV^^}_APPLICATION_ID"
      - name: Publish a/b script
        uses: ./.github/actions/internal-ab
        with:
          nr_environment: ${{ github.event.inputs.env }}
          nrba_app_id: ${{ secrets[steps.local-vars.outputs.app_id_key] }}
          nrba_license_key: ${{ secrets.INTERNAL_LICENSE_KEY }}
          nrba_current_script_url: https://js-agent.newrelic.com/nr-loader-spa-current.min.js
          nrba_next_script_url: https://js-agent.newrelic.com/dev/nr-loader-experimental.min.js
          aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws_role: ${{ secrets.AWS_ROLE_ARN }}
          aws_bucket_name: ${{ secrets.AWS_BUCKET }}
          fastly_key: ${{ secrets.FASTLY_PURGE_KEY }}
```

Comment (patrickhousley) [reply .github/workflows/publish-experiment.yml:1]: Can't. They have different app IDs.

Comment (cwli24) [reply .github/workflows/publish-experiment.yml:1]: Different secrets right? I think addressed that thru another step

**Comment by metal-messiah** [comment]
After discussing with team, i would like to steer this toward a different path.  New backlog tickets have been added, and until then, closing this PR.

**Pull request closed without merging**